app/src/main/java/com/greysonparrelli/permisodemo/MainActivity.java

```diff
@@ -112,36 +114,26 @@ public void onRationaleRequested(Permiso.IOnRationaleProvided callback, String..
      */
     private void onDuplicateClick() {
         // First request
-        Permiso.getInstance().requestPermissions(new Permiso.IOnPermissionResult() {
-            @Override
-            public void onPermissionResult(Permiso.ResultSet resultSet) {
-                if (resultSet.areAllPermissionsGranted()) {
-                    Toast.makeText(MainActivity.this, "Permission Granted! (1)", Toast.LENGTH_SHORT).show();
-                } else {
-                    Toast.makeText(MainActivity.this, "Permission Denied. (1)", Toast.LENGTH_SHORT).show();
-                }
-            }
-
-            @Override
-            public void onRationaleRequested(Permiso.IOnRationaleProvided callback, String... permissions) {
-                Permiso.getInstance().showRationaleInDialog("Permission Rationale", "Needed for demo purposes.", null, callback);
-            }
-        }, Manifest.permission.CAMERA);
+        requestPermissions();
 
         // Second request for the same permission
+        requestPermissions();
+    }
+
+    private void requestPermissions() {
         Permiso.getInstance().requestPermissions(new Permiso.IOnPermissionResult() {
             @Override
             public void onPermissionResult(Permiso.ResultSet resultSet) {
                 if (resultSet.areAllPermissionsGranted()) {
-                    Toast.makeText(MainActivity.this, "Permission Granted! (2)", Toast.LENGTH_SHORT).show();
+                    Toast.makeText(MainActivity.this, "Permission Granted! (1)", Toast.LENGTH_SHORT).show();
```

Comment:
This isn't necessarily equivalent in functionality. It used to be that the first printed a ```(1)``` and the second printed a ```(2)```. We could the same thing by adding an argument to ```requestPermissions()``` that allows it to display different messages.